stdlib/Test/src/Test.jl

```diff
@@ -1305,6 +1305,7 @@ end
     @testset [CustomTestSet] [option=val  ...] ["description"] begin ... end
     @testset [CustomTestSet] [option=val  ...] ["description \$v"] for v in (...) ... end
     @testset [CustomTestSet] [option=val  ...] ["description \$v, \$w"] for v in (...), w in (...) ... end
+    @testset [CustomTestSet] [option=val  ...] ["description \$v, \$w"] let v = (...) ... end
```

Comment:
The `let` variant current deliberately does not support any options (i.e. CustomTestSet, description, etc) to allow room for further extension to give this semantics. Also I thought I had added this. Must have gotten lost in the rebase.

```suggestion
    @testset let v = (...) ... end
```

but probably reorder it to the last case.